code/modules/mob/living/simple_animal/hostile/swarmer.dm

```diff
@@ -0,0 +1,834 @@
+		return
+
+	to_chat(src, "<span class='info'>Attempting to remove this being from our presence.</span>")
+
+	if(!do_mob(src, target, 30))
+		return
+
+	var/turf/open/floor/F
+	F = find_safe_turf(zlevels = z, extended_safety_checks = TRUE)
+
+	if(!F)
+		return
+	// If we're getting rid of a human, slap some energy cuffs on
+	// them to keep them away from us a little longer
+
+	var/mob/living/carbon/human/H = target
+	if(ishuman(target) && (!H.handcuffed))
+		H.handcuffed = new /obj/item/restraints/handcuffs/energy/used(H)
+		H.update_handcuffed()
+		log_combat(src, H, "handcuffed")
+
+	var/datum/effect_system/spark_spread/S = new
+	S.set_up(4,0,get_turf(target))
+	S.start()
+	playsound(src,'sound/effects/sparks4.ogg',50,TRUE)
+	do_teleport(target, F, 0, channel = TELEPORT_CHANNEL_BLUESPACE)
+
+/mob/living/simple_animal/hostile/swarmer/electrocute_act(shock_damage, source, siemens_coeff = 1, flags = NONE)
+	if(!(flags & SHOCK_TESLA))
+		return FALSE
+	return ..()
+
+/**
+  * Called when a swarmer attempts to disassemble a machine
+  *
+  * Proc called when a swarmer attempts to disassemble a machine.  Destroys the machine, and gives the swarmer metal.
+  * Arguments:
+  * * target - The machine the swarmer is attempting to disassemble
+  */
+/mob/living/simple_animal/hostile/swarmer/proc/DismantleMachine(obj/machinery/target)
+	do_attack_animation(target)
+	to_chat(src, "<span class='info'>We begin to dismantle this machine. We will need to be uninterrupted.</span>")
+	var/obj/effect/temp_visual/swarmer/dismantle/D = new /obj/effect/temp_visual/swarmer/dismantle(get_turf(target))
+	D.pixel_x = target.pixel_x
+	D.pixel_y = target.pixel_y
+	D.pixel_z = target.pixel_z
+	if(do_mob(src, target, 100))
+		to_chat(src, "<span class='info'>Dismantling complete.</span>")
+		var/atom/Tsec = target.drop_location()
+		new /obj/item/stack/sheet/metal(Tsec, 5)
+		for(var/obj/item/I in target.component_parts)
+			I.forceMove(Tsec)
+		var/obj/effect/temp_visual/swarmer/disintegration/N = new /obj/effect/temp_visual/swarmer/disintegration(get_turf(target))
+		N.pixel_x = target.pixel_x
+		N.pixel_y = target.pixel_y
+		N.pixel_z = target.pixel_z
+		target.dropContents()
+		if(istype(target, /obj/machinery/computer))
+			var/obj/machinery/computer/C = target
+			if(C.circuit)
+				C.circuit.forceMove(Tsec)
+		qdel(target)
+
+/obj/effect/temp_visual/swarmer //temporary swarmer visual feedback objects
+	icon = 'icons/mob/swarmer.dmi'
+	layer = BELOW_MOB_LAYER
+
+/obj/effect/temp_visual/swarmer/disintegration
+	icon_state = "disintegrate"
+	duration = 10
+
+/obj/effect/temp_visual/swarmer/disintegration/Initialize()
+	. = ..()
+	playsound(loc, "sparks", 100, TRUE)
+
+/obj/effect/temp_visual/swarmer/dismantle
+	icon_state = "dismantle"
+	duration = 25
+
+/obj/effect/temp_visual/swarmer/integrate
+	icon_state = "integrate"
+	duration = 5
+
+/obj/structure/swarmer //Default swarmer effect object visual feedback
```

Comment:
This should be in its own file along with traps blockades and beacons. Can you create a folder for swarmers please